impeller/entity/contents/filters/filter_contents.cc

```diff
@@ -55,23 +56,38 @@ std::shared_ptr<FilterContents> FilterContents::MakeGaussianBlur(
     Sigma sigma_y,
     BlurStyle blur_style,
     Entity::TileMode tile_mode) {
-  std::shared_ptr<FilterContents> x_blur = MakeDirectionalGaussianBlur(
-      /*input=*/input,
-      /*sigma=*/sigma_x,
-      /*direction=*/Point(1, 0),
-      /*blur_style=*/BlurStyle::kNormal,
-      /*tile_mode=*/tile_mode,
-      /*is_second_pass=*/false,
-      /*secondary_sigma=*/{});
-  std::shared_ptr<FilterContents> y_blur = MakeDirectionalGaussianBlur(
-      /*input=*/FilterInput::Make(x_blur),
-      /*sigma=*/sigma_y,
-      /*direction=*/Point(0, 1),
-      /*blur_style=*/blur_style,
-      /*tile_mode=*/tile_mode,
-      /*is_second_pass=*/true,
-      /*secondary_sigma=*/sigma_x);
-  return y_blur;
+  constexpr bool use_new_filter =
+#ifdef IMPELLER_ENABLE_NEW_GAUSSIAN_FILTER
+      true;
+#else
+      false;
+#endif
+
+  // TODO(https://github.com/flutter/flutter/issues/131580): Remove once the new
+  // blur handles all cases.
+  if (use_new_filter) {
+    auto blur = std::make_shared<GaussianBlurFilterContents>(sigma_x.sigma);
+    blur->SetInputs({input});
+    return blur;
+  } else {
+    std::shared_ptr<FilterContents> x_blur = MakeDirectionalGaussianBlur(
```

Comment:
nit: no else if there is an early return.

```suggestion
  if (use_new_filter) {
    auto blur = std::make_shared<GaussianBlurFilterContents>(sigma_x.sigma);
    blur->SetInputs({input});
    return blur;
  }
  
  std::shared_ptr<FilterContents> x_blur = MakeDirectionalGaussianBlur(
```

Reply:
done